packages/amplication-server/src/core/entity/entity.service.ts

```diff
@@ -617,6 +643,26 @@ export class EntityService {
         }
       }
 
+      const resourceWithProject = await this.prisma.resource.findUnique({
+        where: {
+          id: entity.resourceId,
+        },
+        include: {
+          project: true,
+        },
+      });
```

Comment:
I noticed we use `resourceWithProject` in more than one place.
We can add a method to the class

```ts
async resourceWithProject(id: string) {
this.prisma.resource.findUnique({
        where: {
          id,
        },
        include: {
          project: true,
        },
      });}
```

Reply:
and call it with the args id where needed

Reply:
Thanks for the suggestion, I was thinking about it but I don't think it is really beneficial and decided not to

If we are looking to keep things DRY I was thinking about adding the method into LockEntity in the finally clause - but decided to keep it is